src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -249,18 +276,37 @@ class TestRequestRateManager : public TestLoadManagerBase,
     CheckSequences(params_.num_of_sequences);
   }
 
+  /// Test that the shared memory methods are called correctly
+  ///
+  void TestSharedMemory(uint request_rate, uint duration_ms)
+  {
+    ChangeRequestRate(request_rate);
+    std::this_thread::sleep_for(std::chrono::milliseconds(duration_ms));
+    StopWorkerThreads();
+
+    CheckSharedMemory();
+  }
+
+  std::shared_ptr<ModelParser>& parser_{LoadManager::parser_};
+  std::shared_ptr<DataLoader>& data_loader_{LoadManager::data_loader_};
+  bool& using_json_data_{LoadManager::using_json_data_};
+  bool& execute_{RequestRateManager::execute_};
+  std::vector<std::chrono::nanoseconds>& schedule_{
+      RequestRateManager::schedule_};
+
   struct ThreadStat : RequestRateManager::ThreadStat {
   };
-
   struct ThreadConfig : RequestRateManager::ThreadConfig {
     ThreadConfig(uint32_t index, uint32_t stride)
         : RequestRateManager::ThreadConfig(index, stride)
     {
     }
   };
 
-  std::vector<std::chrono::nanoseconds>& schedule_{
-      RequestRateManager::schedule_};
+  void CheckUsingSharedMemoryValue(bool expected_value)
+  {
+    CHECK(expected_value == using_shared_memory_);
```

Comment:
nit: if this is one line, does it need to be its own function? Might be easier to understand the tests if this was explicitly written in the test bodies.

Reply:
You are right. I envisioned this being used more places but I moved it to the tests.